src/main/java/com/avispl/symphony/dal/communicator/aver/ptz/AverPTZCommunicator.java

```diff
@@ -0,0 +1,361 @@
+		String commandString = getHexByteString(command);
+		Iterator<String> iterator = this.getCommandErrorList().iterator();
+
+		String string;
+		do {
+			if (!iterator.hasNext()) {
+				ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+				outputStream.write(Prefix.PAYLOAD_TYPE.getPrefixCode());
+				outputStream.write(PayloadType.INQUIRY.getCode());
+
+				// If type is INQUIRY command, just need to check command error list
+				if (commandString.startsWith(getHexByteString(outputStream.toByteArray()))) {
+					this.logger.trace("Done reading, found inquiry packet reply from: " + this.getHost() + " port: " + this.getPort());
+					return true;
+				}
+
+				iterator = this.getCommandSuccessList().iterator();
+
+				do {
+					if (!iterator.hasNext()) {
+						return false;
+					}
+
+					string = iterator.next();
+				} while (!responseString.endsWith(string));
+
+				if (this.logger.isTraceEnabled()) {
+					this.logger.trace("Done reading, found success string: " + string + " from: " + this.getHost() + " port: " + this.getPort());
+				}
+
+				return true;
+			}
+
+			string = iterator.next();
+		} while (!responseString.endsWith(string));
+
+		if (this.logger.isTraceEnabled()) {
+			this.logger.trace("Done reading, found error string: " + string + " from: " + this.getHost() + " port: " + this.getPort());
+		}
+
+		// if it has ACK packet first -> Error packet, 1 Completion packet left to receive from Device
+		if (hasACK) {
+			super.read(command);
+		}
+
+		throw new CommandFailureException(this.getHost(), commandString, responseString);
+	}
+
+	/**
+	 * This method is used to digest the response received from the device
+	 *
+	 * @param response This is the response to be digested
+	 * @param expectedResponseName This is the expected response string to be compared with received
+	 * @return Object This returns the result digested from the response.
+	 */
+	public Object digestResponse(byte[] response, int sequenceNumber, CommandType commandType, String expectedResponseName) {
+
+		if (response[1] == PayloadType.REPLY.getCode()) {
+			byte[] responseSeqNum = Arrays.copyOfRange(response, 4, 8);
+			byte[] expectedSeqNum = convertIntToByteArray(sequenceNumber);
+
+			if (Arrays.equals(expectedSeqNum, responseSeqNum)) {
+				int payloadLength = response[3];
+				byte[] reply = Arrays.copyOfRange(response, 8, 8 + payloadLength);
+				byte currentValue = reply[2];
+
+				if (commandType == CommandType.COMMAND) {
+					if (!Arrays.equals(ReplyStatus.COMPLETION.getCode(), reply)) {
+						if (this.logger.isErrorEnabled()) {
+							this.logger.error("error: Unexpected completion packet: " + this.host + " port: " + this.port);
+						}
+						throw new RuntimeException("Unexpected completion packet");
+					}
+				} else if (commandType == CommandType.INQUIRY) {
+					switch (expectedResponseName) {
+						case "Power": {
```

Comment:
I see you already defined the name in Command enum. So I think you can parse expectedResponseName to Command enum or you should use Command.POWER.getName() instead of "Power".

Reply:
Ok, I will fix it